types/write-file-webpack-plugin/index.d.ts

```diff
@@ -1,38 +1,42 @@
 import webpack = require("webpack");
 
-export interface UserOptionsType {
-    /**
-     * Atomically replace files content (i.e., to prevent programs like test watchers from seeing partial files).
-     * @default true
-     */
-    atomicReplace?: boolean | undefined;
-    /**
-     * Stop writing files on webpack errors
-     * @default true
-     */
-    exitOnErrors?: boolean | undefined;
-    /**
-     * A regular expression or function used to test if file should be written.
-     * When not present, all bundle will be written.
-     */
-    test?: RegExp | undefined;
-    /**
-     * Use hash index to write only files that have changed since the last iteration.
-     * @default true
-     */
-    useHashIndex?: boolean | undefined;
-    /**
-     * Logs names of the files that are being written (or skipped because they have not changed)
-     * @default true
-     */
-    log?: boolean | undefined;
-    /**
-     * Forces the execution of the plugin regardless of being using `webpack-dev-server` or not
-     * @default false
-     */
-    force?: boolean | undefined;
+declare namespace WriteFilePlugin {
+    interface UserOptionsType {
+        /**
+         * Atomically replace files content (i.e., to prevent programs like test watchers from seeing partial files).
+         * @default true
+         */
+        atomicReplace?: boolean | undefined;
+        /**
+         * Stop writing files on webpack errors
+         * @default true
+         */
+        exitOnErrors?: boolean | undefined;
+        /**
+         * A regular expression or function used to test if file should be written.
+         * When not present, all bundle will be written.
+         */
+        test?: RegExp | undefined;
+        /**
+         * Use hash index to write only files that have changed since the last iteration.
+         * @default true
+         */
+        useHashIndex?: boolean | undefined;
+        /**
+         * Logs names of the files that are being written (or skipped because they have not changed)
+         * @default true
+         */
+        log?: boolean | undefined;
+        /**
+         * Forces the execution of the plugin regardless of being using `webpack-dev-server` or not
+         * @default false
+         */
+        force?: boolean | undefined;
+    }
 }
 
-export default class WriteFilePlugin extends webpack.Plugin {
-    constructor(userOptions?: UserOptionsType);
+declare class WriteFilePlugin extends webpack.Plugin {
+    constructor(userOptions?: WriteFilePlugin.UserOptionsType);
 }
+
+export = WriteFilePlugin;
```

Comment:
Not sure that this is correct. The transpiled code (see https://www.npmjs.com/package/write-file-webpack-plugin?activeTab=code) for this plugin does define `exports.default` and defines `exports.__esModule` as `true`. `module.exports` is set also, but that's only for compatibility. I believe the intention is for this to be treated as an ESM default import when using TypeScript

Reply:
No, this is accurate. `module.exports = ...` is exactly what `export = ` represents in TypeScript. The actual module received is not equivalent to a module object, it is the export itself.

Reply:
Per the way this is used in https://www.npmjs.com/package/write-file-webpack-plugin?activeTab=code, setting `exports.default` and `exports.__esModule` doesn't matter as that object is discarded when `module.exports = exports['default']` is evaluated. It's essentially the same as writing `module.exports = WriteFileWebpackPlugin;`

Reply:
Hi @rbuckton , in this case would you like to help giving approval again? Or do I need to repush the PR to remove the request change label first?